data/templates/secure_erase.py

```diff
@@ -56,6 +65,209 @@
 HDPARM_RETRY_EXITCODES = [5]
 COMMAND_LOG_MARKER = "\n==========================" \
                      "==========================\n"
+if os.getcwd()[-1] != '/':
+    PATH = os.getcwd() + "/"
+else:
+    PATH = os.getcwd()
+
+
+class Progress:
+    """
+    Secure erase job progress class.
+    """
+
+    def __init__(self, disks, path, parameters):
+        self.parameters = {                     # parameters parsed from command
+            "taskId": parameters["taskId"],     # task id for notification
+            "address": parameters["address"],   # rackhd server address
+            "tool": parameters["tool"],         # erase tool to be used
+            "option": parameters["option"]      # erase command arguments options
+        }
+        self.disk_list = disks  # disks to be erased
+        self.interval = 10      # erase progress polling interval
+        self.percent = 0.0      # last progress percent buffer
+        self.duration = {}      # erase duration for each disks
+        self.path = path        # log file path
+
+    def scrub_parser(self, drive):
```

Comment:
Confusing function name. The name of the function seems like a string or output parser but it's actually a command runner AND a parser. As a reference, the ipmi-parser function in RackHD (https://github.com/RackHD/on-tasks/blob/master/lib/utils/job-utils/ipmi-parser.js) is a pure parser who took the command stdout string as input and decoder it into other format.  Same happen on the other parser functions in the same file.